src/coreclr/jit/fgopt.cpp

```diff
@@ -1871,20 +1871,23 @@ bool Compiler::fgOptimizeSwitchBranches(BasicBlock* block)
                 }
             }
 
-            // Update the switch jump table (this has to happen before calling UpdateSwitchTableTarget)
+            // Update the switch jump table
             FlowEdge* const newEdge = fgAddRefPred(bNewDest, block, fgRemoveRefPred(bDest, block));
             *jmpTab                 = newEdge;
 
-            // Maintain, if necessary, the set of unique targets of "block."
-            UpdateSwitchTableTarget(block, bDest, bNewDest);
-
             // we optimized a Switch label - goto REPEAT_SWITCH to follow this new jump
-            returnvalue = true;
+            modified = true;
 
             goto REPEAT_SWITCH;
         }
     } while (++jmpTab, --jmpCnt);
 
+    if (modified)
+    {
+        // Invalidate the set of unique targets for block, since we modified the targets
+        InvalidateUniqueSwitchSuccMap();
```

Comment:
```suggestion
        fgInvalidateUniqueSwitchSuccMapEntry(block);
```
? Or do we need to invalidate the full thing?

Reply:
That seems to work; fixed.